R/get_feature_wfs.R

```diff
@@ -81,57 +90,100 @@ get_feature_wfs <- function(
       bbox[["xmax"]],
       bbox[["ymax"]],
-      sep = ",")
+      sep = ","
+    )
   }
   if (grepl(pattern = "^2", x = version)) {
-    url$query <- list(service = "wfs",
-                      request = "GetFeature",
-                      version = version,
-                      typeNames = layername,
-                      srsName = crs,
-                      bbox = bbox,
-                      filter = filter,
-                      cql_filter = cql_filter,
-                      outputFormat = output_format,
-                      propertyName = property_name,
-                      resultType = result_type,
-                      ...
+    url$query <- list(
+      service = "wfs",
+      request = "GetFeature",
+      version = version,
+      typeNames = layername,
+      srsName = crs,
+      bbox = bbox,
+      filter = filter,
+      cql_filter = cql_filter,
+      outputFormat = output_format,
+      propertyName = property_name,
+      resultType = result_type,
+      ...
     )
   }
   if (grepl(pattern = "^1", x = version)) {
-    url$query <- list(service = "wfs",
-                      request = "GetFeature",
-                      version = version,
-                      typeName = layername,
-                      srsName = crs,
-                      bbox = bbox,
-                      filter = filter,
-                      cql_filter = cql_filter,
-                      outputFormat = output_format,
-                      propertyName = property_name,
-                      resultType = NULL,
-                      ...
+    url$query <- list(
+      service = "wfs",
+      request = "GetFeature",
+      version = version,
+      typeName = layername,
+      srsName = crs,
+      bbox = bbox,
+      filter = filter,
+      cql_filter = cql_filter,
+      outputFormat = output_format,
+      propertyName = property_name,
+      resultType = NULL,
+      ...
     )
   }
 
   request <- build_url(url)
 
-  result <- GET(request)
-  parsed <- as_list(content(result, "parsed", encoding = "UTF-8"))
-
-  if (names(parsed) == "ExceptionReport") {
-    message <- unlist(parsed$ExceptionReport$Exception$ExceptionText)
-    old_op <- options(warning.length = max(nchar(message), 1000))
-    on.exit(options(old_op))
-    stop(sprintf(paste0(message, "\nThe requested url was: %s"),
-                 request))
-  }
+  get_result <- GET(request)
+  handle_result_types(
+    get_result,
+    result_type = result_type,
+    property_name = property_name,
+    request = request
+  )
+}
 
-  if (result_type == "hits") {
-    n_features <- attr(parsed$FeatureCollection, "numberMatched")
-    return(n_features)
+handle_result_types <- function(result, result_type, property_name, request) {
+  if (result$status_code == 200L) {
+    if (result_type == "hits") {
+      parsed <- as_list(content(result, "parsed", encoding = "UTF-8"))
+      n_features <- attr(parsed$FeatureCollection, "numberMatched")
+      return(n_features)
+    } else {
```

Comment:
You can omit the else when using early returns. That makes the code IMHO easier to read. Flipping the test makes the code shorter too and brings short exception forward.

Example
```
if (testA == "A") {
  if (testB == "a") {
    return("Aa")
  } else {
    return("Ab")
  }
} else {
  return("B")
}
```
Alternative version
```
if (testA != "A") {
  return("B")
}
if (testB == "a") {
  return("Aa")
}
return("Ab")
```

Reply:
see [efa54d6](https://github.com/inbo/inbospatial/pull/1/commits/efa54d606d348cbb4514f97dbd014021634ce692)